docs/aws/audit/lambdamonitoring/rules/multiple_functions_with_same_iam.mdx

```diff
@@ -23,6 +23,243 @@ SOC2
+5. Repeat steps 3 and 4 for all the functions in your list. If you find that multiple functions are using the same IAM role, then you have detected a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the Lambda functions in your AWS account. You can do this by using the AWS CLI command `aws lambda list-functions`. This command will return a JSON object with all your Lambda functions.
+
+    ```
+    aws lambda list-functions --region your-region
+    ```
+
+2. Next, you need to extract the IAM role for each Lambda function. You can do this by parsing the JSON object returned by the previous command. Each function's details are stored in a dictionary, and the IAM role can be found under the 'Role' key.
+
+    ```
+    aws lambda list-functions --region your-region --query 'Functions[*].[FunctionName,Role]' --output text
+    ```
+
+3. Now, you need to check if multiple functions are using the same IAM role. You can do this by storing the IAM roles in a list and then checking if any role appears more than once in the list. This can be done using a Python script. Here is a simple script that does this:
+
+    ```python
+    import subprocess
+    import collections
+
+    command = "aws lambda list-functions --region your-region --query 'Functions[*].[FunctionName,Role]' --output text"
+    output = subprocess.check_output(command, shell=True).decode('utf-8').split('\n')
+
+    roles = [line.split('\t')[1] for line in output if line]
+    duplicates = [item for item, count in collections.Counter(roles).items() if count > 1]
+
+    if duplicates:
+        print(f"The following IAM roles are used by multiple Lambda functions: {', '.join(duplicates)}")
+    else:
+        print("No IAM roles are used by multiple Lambda functions.")
+    ```
+
+4. Run the Python script. If there are any IAM roles used by multiple Lambda functions, they will be printed out. If not, a message will be printed saying that no IAM roles are used by multiple Lambda functions.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries and establish a session with AWS using Boto3. Boto3 is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, and others.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+```
+
+2. Create a client for AWS Lambda and IAM. 
+
+```python
+lambda_client = session.client('lambda')
+iam_client = session.client('iam')
+```
+
+3. Get a list of all the Lambda functions and their respective IAM roles. 
+
+```python
+functions = lambda_client.list_functions()['Functions']
+role_dict = {}
+
+for function in functions:
+    role = function['Role']
+    if role not in role_dict:
+        role_dict[role] = [function['FunctionName']]
+    else:
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>